spec/support/matchers.rb

```diff
@@ -19,3 +19,28 @@ def body_params(request)
     Oj.load(request.encoded_body)
   end
 end
+
+RSpec::Matchers.define :have_query_params do |params|
+  match do |request|
+    body = fetch_uri_params(request)
+
+    values_match?(params, body)
+  end
+  failure_message do |request|
+    super() + ", but had #{fetch_uri_params(request)}"
+  end
+  failure_message_when_negated do |request|
+    super() + ", but had #{fetch_uri_params(request)}"
+  end
+
+  private
+
+  # TODO: This method is also used in spec/doubles/base.rb. Where can we put it
```

Comment:
Let's talk about this tomorrow. This method is small enough that having it defined twice in each logical group is fine by me. Will still already merge it now since this isn't a blocker from my PoV